MONGOID-4528 Add more dirty methods (#5440)
mongodb/mongoid: docs/release-notes/mongoid-8.1.txt

------------------------------------------------------------------------------------------------------------------

These new methods behave identically to corresponding methods
from ``ActiveRecord::AttributeMethods::Dirty``. The methods are particularly useful in
callbacks:

.. code-block:: ruby

  class Person
    include Mongoid::Document

    field :name, type: String

    before_save do
      puts "will_save_change_to_attribute?(:name): #{will_save_change_to_attribute?(:name)}"
    end

    after_save do
      puts "attribute_was(:name): #{attribute_was(:name)}"
      puts "attribute_before_last_save(:name): #{attribute_before_last_save(:name)}"
      puts "saved_change_to_attribute(:name): #{saved_change_to_attribute(:name)}"
    end
  end

  person = Person.create(name: 'John')
  # "will_save_change_to_attribute?(:name): true"
  # attribute_was(:name): John => New value
  # attribute_before_last_save(:name): nil => Value before save
  # saved_change_to_attribute(:name): [nil, "John"] => Both values
  person.name = 'Jane'
  person.save
  # "will_save_change_to_attribute?(:name): true"

>   # attribute_was(:name): Jane => New value

Reviewer: does attribute_was not look back before the last save? if not, can you mention that here?

Author: Done

  # attribute_before_last_save(:name): John => Value before save
  # saved_change_to_attribute(:name): ["John", "Jane"] => Both values

> For all of the new method there is also a shorter form created dynamically, e.g.

Reviewer 2: ```suggestion
For all of the new methods there are also shorter forms created dynamically, e.g.
```
